futures-test/src/assert.rs

```diff
@@ -2,16 +2,38 @@ use futures_core::stream::Stream;
 use std::marker::Unpin;
 
 #[doc(hidden)]
-pub fn assert_is_unpin_stream<S: Stream + Unpin>(_: &S) {}
+pub fn assert_is_unpin_stream<S: Stream + Unpin>(_: &mut S) {}
 
 /// Assert that the next poll to the provided stream will return
 /// [`Poll::Pending`][futures_core::task::Poll::Pending].
+///
+/// # Examples
+///
+/// ```
+/// #![feature(async_await, futures_api, pin)]
+///
+/// #[macro_use]
+/// extern crate futures_test;
+///
+/// #[macro_use]
+/// extern crate futures;
+///
+/// # fn main() {
+/// use futures::stream;
+/// use futures_test::future::FutureTestExt;
+///
+/// let mut stream = stream::once((async { 5 }).delay());
+/// pin_mut!(stream);
+///
+/// assert_stream_pending!(stream);
```

Comment:
It'd be good if the examples showed that multiple consecutive calls to these assert function are possible.